### PR TITLE
[X86][BF16] Support INSERT_SUBVECTOR and CONCAT_VECTORS

### PR DESCRIPTION
**Pull request by phoebewang**

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrSSE.td [hunk 1]
```diff
@@ -7931,6 +7927,14 @@ let Predicates = [HasAVX2, NoVLX] in {
   defm : vextract_lowering<"VEXTRACTI128", v32i8,  v16i8>;
 }
 
+let Predicates = [HasAVXNECONVERT, NoVLX] in {
+  def : Pat<(v16bf16 (X86SubVBroadcastld128 addr:$src)),
```
Comment (FreddyLeaf) [inline llvm/lib/Target/X86/X86InstrSSE.td:7931]: This file seems to have a good organization on categories of broadcast, insert and extract. Why put these three together?

File: llvm/lib/Target/X86/X86InstrSSE.td [hunk 2]
```diff
@@ -7931,6 +7927,14 @@ let Predicates = [HasAVX2, NoVLX] in {
...
+            (VBROADCASTF128rm addr:$src)>;
+  defm : vinsert_lowering<"VINSERTI128", "VPERM2I128", v8bf16,
+                          v16bf16, loadv8bf16,  loadv16bf16>;
+  defm : vextract_lowering<"VEXTRACTI128", v16bf16, v8bf16>;
```
Comment (FreddyLeaf) [inline llvm/lib/Target/X86/X86InstrSSE.td:7935]: do we need to add tests for extract?

Comment (phoebewang) [reply llvm/lib/Target/X86/X86InstrSSE.td:7935]: Done.

File: llvm/lib/Target/X86/X86InstrSSE.td (hunk 1 above)
Comment (phoebewang) [reply llvm/lib/Target/X86/X86InstrSSE.td:7931]: In that way, we need to check Predicates each time. I'm fine with either way. I can change it if you prefer.

Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86InstrSSE.td:7931]: Yes, I prefer to the old way.

Comment (phoebewang) [reply llvm/lib/Target/X86/X86InstrSSE.td]: Done.

**Review by FreddyLeaf — APPROVED** [review]
LGTM

**Comment by phoebewang** [comment]
Thanks @FreddyLeaf !

File: llvm/test/CodeGen/X86/avx512bf16-vl-intrinsics.ll [hunk 3]
```diff
@@ -372,31 +372,11 @@ entry:
 
 ;; FIXME: This should generate the same output as above, but let's fix the crash first.
```
Comment (RKSimon) [inline llvm/test/CodeGen/X86/avx512bf16-vl-intrinsics.ll:373]: Remove FIXME?

Comment (phoebewang) [reply llvm/test/CodeGen/X86/avx512bf16-vl-intrinsics.ll:373]: Good catch! Removed by https://github.com/llvm/llvm-project/commit/6c87f46795699ee0997ebb85365d9df45e48292c

**Pull request merged**